### PR TITLE
Batched API update

### PR DESCRIPTION
**Pull request by mratsim**

### CODE REVIEW

File: blscurve/bls_batch_verifier.nim [hunk 1]
```diff
@@ -31,116 +31,60 @@ import
...
+  if input.len == 0:
     # Spec precondition
     return false
 
-  batcher.batchContexts.setLen(1)
-  template ctx: untyped = batcher.batchContexts[0]
-  batcher.batchContexts[0].init(secureRandomBytes, "")
+  cache.batchContexts.setLen(1)
+  template ctx: untyped = cache.batchContexts[0]
+  cache.batchContexts[0].init(secureRandomBytes, "")
```
Comment (arnetheduck) [inline blscurve/bls_batch_verifier.nim:80]:
```suggestion
  ctx.init(secureRandomBytes, "")
```

File: blscurve/bls_batch_verifier.nim [hunk 2]
```diff
@@ -31,116 +31,60 @@ import
...
 
   # Accumulate line functions
-  for i in 0 ..< batcher.sets.len:
+  for i in 0 ..< input.len:
```
Comment (arnetheduck) [inline blscurve/bls_batch_verifier.nim:83]: why not `for i in input`?

File: blscurve/bls_batch_verifier.nim [hunk 3]
```diff
@@ -280,29 +241,37 @@ checksAndStackTracesOff:
...
   if numSets == 0:
     # Spec precondition
     return false
 
   let numBatches = min(numSets, omp_get_max_threads().uint32)
 
   # Stage 0: Accumulators - setLen for noinit of seq
-  batcher.batchContexts.setLen(numBatches.int)
-  batcher.updateResults.setLen(numBatches.int)
+  cache.batchContexts.setLen(numBatches.int)
```
Comment (arnetheduck) [inline blscurve/bls_batch_verifier.nim:265]: if contexts.len > numbatches.len, should we shrink it? it keep it?

**Review by arnetheduck — APPROVED** [review]

File: blscurve/bls_batch_verifier.nim (hunk 2 above)
Comment (mratsim) [reply blscurve/bls_batch_verifier.nim:83]: I don't want to put 96 (PublicKey) + 16 (seq[byte]) + 192 (Signature) bytes on the stack.
https://github.com/nim-lang/Nim/issues/14421

File: blscurve/bls_batch_verifier.nim (hunk 3 above)
Comment (mratsim) [reply blscurve/bls_batch_verifier.nim:265]: `setLen` will shrink it (or extend it and reallocate it if needed)

Comment (arnetheduck) [reply blscurve/bls_batch_verifier.nim:265]: yes, but we could potentially skip the shrinking

Comment (mratsim) [reply blscurve/bls_batch_verifier.nim:265]: In that case the reserved memory stays the same and the code does only `mySeq.len = numBatches`.

https://github.com/nim-lang/Nim/blob/0956a99/lib/system/seqs_v2.nim#L85 (ARC seqs but I think the normal seq are the same).

Comment (arnetheduck) [reply blscurve/bls_batch_verifier.nim:265]: sort of, it reruns the zero-init on regrowth

**Pull request merged**